grails-app/controllers/org/pih/warehouse/api/putaway/PutawayTaskApiController.groovy

```diff
@@ -0,0 +1,84 @@
+package org.pih.warehouse.api.putaway
+
+import grails.converters.JSON
+import org.pih.warehouse.api.StatusCategory
+import org.pih.warehouse.core.Location
+import org.pih.warehouse.product.Product
+import org.pih.warehouse.putaway.PutawayTask
+import org.pih.warehouse.putaway.PutawayTaskService
+import org.springframework.http.HttpStatus
+
+class PutawayTaskApiController {
+
+    static responseFormats = ['json']
+    static allowedMethods = [
+            index: 'GET',
+            show: 'GET',
+            save: 'POST',
+            update: 'PUT',
+            patch: ['PATCH', 'POST'],
+            delete: 'DELETE',
+            generateForOrder: 'POST'
+    ]
+
+    PutawayTaskService putawayTaskService
+
+    def search(SearchPutawayTaskCommand command) {
+        log.info "search params " + params
+        // Return a not found error if request specifies a product, but the product does not exist
+        if (command.product && !command.product) {
```

Comment:
It could be useful to allow sending a product code or upc, and do a product search here too (if no product in the payload). Right now we have to hit the barcode product search api, and then this one. But it's fine if you want to keep it separated.